cpp/src/groupby/sort/group_tdigest.cu

```diff
@@ -0,0 +1,849 @@
+    // the range of tdigests in this group
+    auto const tdigest_start = h_outer_offsets[outer_index];
+    auto const tdigest_end   = h_outer_offsets[outer_index + 1];
+    auto const num_tdigests  = tdigest_end - tdigest_start;
+
+    // slice each tdigest from the input
+    std::vector<table_view> unmerged_tdigests;
+    unmerged_tdigests.reserve(num_tdigests);
+    auto offset_iter = thrust::make_counting_iterator(tdigest_start);
+    std::transform(offset_iter,
+                   offset_iter + num_tdigests,
+                   std::back_inserter(unmerged_tdigests),
+                   [&](int inner_index) {
+                     std::vector<size_type> indices{h_inner_offsets[inner_index],
+                                                    h_inner_offsets[inner_index + 1]};
+                     return cudf::detail::slice(tdigests_unsliced, indices, stream);
+                   });
+
+    // merge
+    return cudf::detail::merge(unmerged_tdigests, {0}, {order::ASCENDING}, {}, stream, mr);
+  });
+
+  // generate min and max values
+  auto min_col        = scv.child(cudf::detail::tdigest::min_column_index);
+  auto merged_min_col = cudf::make_fixed_width_column(
+    data_type{type_id::FLOAT64}, num_groups, mask_state::UNALLOCATED, stream, mr);
+  thrust::reduce_by_key(rmm::exec_policy(stream),
+                        group_labels.begin(),
+                        group_labels.end(),
+                        min_col.begin<double>(),
+                        thrust::make_discard_iterator(),
+                        merged_min_col->mutable_view().begin<double>(),
+                        thrust::equal_to<size_type>{},  // key equality check
+                        thrust::minimum<double>{});
+
+  auto max_col        = scv.child(cudf::detail::tdigest::max_column_index);
+  auto merged_max_col = cudf::make_fixed_width_column(
+    data_type{type_id::FLOAT64}, num_groups, mask_state::UNALLOCATED, stream, mr);
+  thrust::reduce_by_key(rmm::exec_policy(stream),
+                        group_labels.begin(),
+                        group_labels.end(),
+                        max_col.begin<double>(),
+                        thrust::make_discard_iterator(),
+                        merged_max_col->mutable_view().begin<double>(),
+                        thrust::equal_to<size_type>{},  // key equality check
+                        thrust::maximum<double>{});
+
+  // concatenate all the merged tdigests back into one table.
+  std::vector<table_view> tdigest_views;
+  tdigest_views.reserve(num_groups);
+  std::transform(tdigests.begin(),
+                 tdigests.end(),
+                 std::back_inserter(tdigest_views),
+                 [](std::unique_ptr<table> const& t) { return t->view(); });
+  auto merged = cudf::detail::concatenate(tdigest_views, stream, mr);
+
+  // generate cumulative weights
+  auto cumulative_weights =
+    std::make_unique<column>(merged->get_column(cudf::detail::tdigest::weight_column_index),
```

Comment:
Is this intentionally making a copy?

Reply:
Yes.  There's an inclusive_scan that gets applied to it below.

Reply:
Oh, you mean:  just make an empty column and scan from the weights column. Right.